services/billing.py

```diff
@@ -204,39 +204,42 @@ def modify_subscription(self, owner, desired_plan):
 
         if is_upgrading:
             if subscription_schedule_id:
-                self._modify_subscription_schedule(
-                    owner, subscription, subscription_schedule_id, desired_plan
-                )
-            else:
                 log.info(
-                    f"Updating Stripe subscription for owner {owner.ownerid} to {desired_plan['value']} by user #{self.requesting_user.ownerid}"
-                )
-                stripe.Subscription.modify(
-                    owner.stripe_subscription_id,
-                    cancel_at_period_end=False,
-                    items=[
-                        {
-                            "id": subscription["items"]["data"][0]["id"],
-                            "plan": settings.STRIPE_PLAN_IDS[desired_plan["value"]],
-                            "quantity": desired_plan["quantity"],
-                        }
-                    ],
-                    metadata=self._get_checkout_session_and_subscription_metadata(
-                        owner
-                    ),
-                    proration_behavior=proration_behavior,
+                    f"Releasing Stripe schedule for owner {owner.ownerid} to {desired_plan['value']} with {desired_plan['quantity']} seats by user #{self.requesting_user.ownerid}"
                 )
+                stripe.SubscriptionSchedule.release(subscription_schedule_id)
+            log.info(
+                f"Updating Stripe subscription for owner {owner.ownerid} to {desired_plan['value']} by user #{self.requesting_user.ownerid}"
+            )
+            stripe.Subscription.modify(
+                owner.stripe_subscription_id,
+                cancel_at_period_end=False,
+                items=[
+                    {
+                        "id": subscription["items"]["data"][0]["id"],
+                        "plan": settings.STRIPE_PLAN_IDS[desired_plan["value"]],
+                        "quantity": desired_plan["quantity"],
+                    }
+                ],
+                metadata=self._get_checkout_session_and_subscription_metadata(owner),
+                proration_behavior=proration_behavior,
+            )
 
-                plan_service = PlanService(current_org=owner)
-                plan_service.update_plan(
-                    name=desired_plan["value"], user_count=desired_plan["quantity"]
-                )
+            plan_service = PlanService(current_org=owner)
+            plan_service.update_plan(
+                name=desired_plan["value"], user_count=desired_plan["quantity"]
+            )
 
-                log.info(
-                    f"Stripe subscription modified successfully for owner {owner.ownerid} by user #{self.requesting_user.ownerid}"
-                )
+            log.info(
+                f"Stripe subscription modified successfully for owner {owner.ownerid} by user #{self.requesting_user.ownerid}"
+            )
         else:
```

Comment:
under this else block can be simplified to:
```
if not subscription_schedule_id:
    subscription_schedule_id = stripe.SubscriptionSchedule.create(
        from_subscription=owner.stripe_subscription_id
    ).id

self._modify_subscription_schedule(
    owner, subscription, subscription_schedule_id, desired_plan
)
```

Reply:
makes sense, will make the changes!